src/lando/utils/github.py

```diff
@@ -220,3 +265,101 @@ def serialize(self) -> dict[str, str]:
             "user_html_url": self.user_html_url,
             "user_login": self.user_login,
         }
+
+    def get_diff(self, client: GitHubAPIClient) -> str:
+        """Return a single diff of the latest state for the PR.
+
+        WARNING: The returned diff doesn't include any binary data.
+
+        If Binary data is desired, the `get_patch` method should be used instead.
+        """
+        response = client.session.get(self.diff_url)
+        response.raise_for_status()
+
+        return response.text
+
+    def get_patch(self, client: GitHubAPIClient) -> str:
+        """Return a series of patches from the PR's commits.
+
+        Patches from each commit are concatenated into a single string.
+
+        This includes binary content, unlike `get_diff`.
+        """
+        response = client.session.get(self.patch_url)
+        response.raise_for_status()
+
+        return response.text
+
+
+class PullRequestPatchHelper(PatchHelper):
+    """A PatchHelper-like wrapper for GitHub pull requests.
+
+    Due to the nature of pull requests, it only implement the data-getting
+    functionality, and  doesn't implement the input and output
+    methods.
+    """
+
+    _diff: str
+
+    def __init__(self, client: GitHubAPIClient, pr: PullRequest):
+        super().__init__()
+
+        self._diff = pr.get_diff(client)
+
+        user = f"{pr.user_login}@github-pr"
+        # Consider the committer of the first patch to be the author.
+        patch = pr.get_patch(client)
+        for line in patch.splitlines():
+            if match := self.USERNAME_RE.match(line):
+                user = match["user"]
+                break
```

Comment:
I think this should be delegated to a method on the `PullRequest` object, since the behaviour will change when [bug 1995006](https://bugzilla.mozilla.org/show_bug.cgi?id=1995006) is implemented.